mycore-restapi/src/main/java/org/mycore/restapi/v1/utils/MCRRestAPIUploadHelper.java

```diff
@@ -114,21 +116,15 @@ public static Response uploadObject(UriInfo info, HttpServletRequest request, In
             Document docOut = sb.build(uploadedInputStream);
 
             MCRObjectID mcrID = MCRObjectID.getInstance(docOut.getRootElement().getAttributeValue("ID"));
-            if (mcrID.getNumberAsInteger() == 0) {
-                mcrID = MCRObjectID.getNextFreeId(mcrID.getBase());
-            }
-
-            fXML = UPLOAD_DIR.resolve(mcrID + ".xml");
 
-            docOut.getRootElement().setAttribute("ID", mcrID.toString());
-            docOut.getRootElement().setAttribute("label", mcrID.toString());
+            fXML = UPLOAD_DIR.resolve(mcrID.getBase() + '_' + UUID.randomUUID() + ".xml");
             XMLOutputter xmlOut = new XMLOutputter(Format.getPrettyFormat());
             try (BufferedWriter bw = Files.newBufferedWriter(fXML, StandardCharsets.UTF_8)) {
                 xmlOut.output(docOut, bw);
             }
 
-            MCRObjectCommands.updateFromFile(fXML.toString(), false); // handles "create" as well
-
+            MCRObject object = MCRObjectCommands.updateFromFile(fXML.toString(), false);// handles "create" as well
+            mcrID = Objects.requireNonNull(object, "Created object should be not null").getId();
```

Comment:
better something like "Object could not be created" / "an error occured while the object was created"
because if thhe value is null something in updateFromFile went wrong

Reply:
This is done in the catch-block.